integration_tests/src/main/python/window_function_test.py

```diff
@@ -213,3 +213,53 @@ def test_window_aggs_for_ranges_of_dates(data_gen):
         '       range between 1 preceding and 1 following) as sum_c_asc '
         'from window_agg_table'
     )
+
+
+'''
+  Spark will drop nulls when collecting, but seems GPU does not, so exceptions come up.
```

Comment:
We cannot merge this in as is. We missed a fundamental requirement when implementing this for cudf.  The best we can do for now is check if the column is nullable or not.  But Spark does not always calculate this properly, because it is being cautious. So we should just wait on this.

Reply:
I spoke with @sameerz about this and I think we can work around this for the 0.4 release.

We can check the input column to see if it is nullable or not.  If it is not nullable we can replace the collect list expression.  If it is nullable then we can check a config to see if we force it to run the collect list anyways.  Or we can just have collect_list off by default and let users enable it if they want this functionality before it is 100% correct.

I would like us to modify the tests with this so that in the short term we have a specific test for collect_list separate from the others so we don't stop testing nulls to enable this.

Reply:
Thanks for review, and I chose the later one to disable it by default.

Sorry I don't fully get your suggestion on how to modify the integration tests, could you pls explain more ?